### PR TITLE
feat(tooltip-signpost): added tooltip and signpost

### PR DESCRIPTION
**Pull request by raizen62**
Tooltip : 
psHKXm6OJX
Signpost:
61VoCuOOh0

### CODE REVIEW

File: libs/angular-components/src/signpost/src/signpost.component.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<button id="signpost" mat-icon-button>
+  <mat-icon>info</mat-icon>
```
Comment (rbordeanu) [inline libs/angular-components/src/signpost/src/signpost.component.html]: let's use the `info_outlined` icon, I think it looks better

File: libs/angular-components/src/signpost/src/signpost.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<button id="signpost" mat-icon-button>
```
Comment (rbordeanu) [inline libs/angular-components/src/signpost/src/signpost.component.html]: Not sure if it should be a button, or a more "discrete" element: i.e simple div with primary color on hover?

File: libs/angular-components/src/tooltip/src/tooltip.directive.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    this.loadTippy();
+  }
+
+  loadTippy() {
+    setTimeout(() => {
+      const el = this.el.nativeElement;
+      console.log(el);
+      const tippyOptions = {
+        placement: this.position || 'top',
+        content: this.content || 'Default tooltip'
```
Comment (rbordeanu) [inline libs/angular-components/src/tooltip/src/tooltip.directive.ts]: Should not have default content, throw an error if content is not provided.

File: libs/angular-components/src/signpost/src/signpost.component.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  @Input() content;
+  @Input() position;
+  ngOnInit() {
+    this.loadTippy();
+  }
+
+  loadTippy() {
+    setTimeout(() => {
+      const tippyOptions =  {
+        content: this.content || 'Hello',
```
Comment (rbordeanu) [inline libs/angular-components/src/signpost/src/signpost.component.ts]: Should not have default content, throw an error if content is not provided.

File: libs/angular-components/src/signpost/src/signpost.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        placement: this.position || 'right',
```
Comment (rbordeanu) [inline libs/angular-components/src/signpost/src/signpost.component.ts]: Should default to 'top'.

File: libs/angular-components/src/signpost/src/signpost.component.ts (hunk 4 above)
Comment (rbordeanu) [reply libs/angular-components/src/signpost/src/signpost.component.ts]: Also, make sure that content can have safe html, like clickable links.

File: themes/angular-theme/lib/tooltip/_tooltip-theme.scss [hunk 6]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (rbordeanu) [inline themes/angular-theme/lib/tooltip/_tooltip-theme.scss]: Don't hardcode color values or anything :) 
Please see best practices for defining a component theme: https://github.com/fusionfabric/finastra-design-system/blob/master/themes/angular-theme/lib/button/_button-theme.scss#L2
(for example)
Define a mixin, use custom variables, extract known values from theme object etc.

File: libs/angular-components/src/signpost/package.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "@ffdc/uxg-angular-components/tooltip",
+  "version": "0.0.1"
```
Comment (aifrim) [inline libs/angular-components/src/signpost/package.json]: Due to recent changes you will need to manually update version to `0.12.0`

File: themes/angular-theme/lib/core/theming/_all-theme.scss [hunk 8]
```diff
@@ -11,6 +11,7 @@
 @import '../../menu/menu-theme';
 @import '../../sidenav/sidenav-theme';
 @import '../../toolbar/toolbar-theme';
+@import '../../tooltip/tooltip-theme';
```
Comment (aifrim) [inline themes/angular-theme/lib/core/theming/_all-theme.scss]: We will need to remove this. @ttalbot can give you more information.

File: themes/angular-theme/lib/tooltip/_tooltip-theme.scss (hunk 6 above)
Comment (aifrim) [reply themes/angular-theme/lib/tooltip/_tooltip-theme.scss]: This file needs to be removed from the theme and added to the component.

**Comment by aifrim** [comment]
> My opinion is that both tooltip and signpost should be part of the theme and based on Angular-Material tooltip.

We should discuss this during the next FDS Sync

**Comment by rbordeanu** [comment]
> My opinion is that both tooltip and signpost should be part of the theme and based on Angular-Material tooltip.

Well I disagree with that.

For once, the Material tooltip is very limiting.
As per spec: 
_Tooltips don’t display rich information including images and formatted text.
Tooltips are different than ALT-attributes, which are intended primarily for static images.
Tooltips don’t have directional arrows; instead, they rely on motion emanating from the source to convey direction._

We DO want to have rich content (HTML) inside the signpost popover, as well as the arrows indicating origin.

Tippy is a really small dependency, with many customisable options, and it looks great (subjective opinion but quite popular).

We'd like to have ready-made directives handling tooltips and signposts to be used directly by developers in a consistent way, without any extra work: like `uxgTooltip` and `uxgSignpost content="whatever"`.

**Comment by aifrim** [comment]
Closing this PR because it is obsolete.

**Pull request closed without merging**